### PR TITLE
Improve parser fallback behavior and allow configurable local runner image

### LINKED ISSUE #96: Fallback when workflow parser omits jobs + configurable local runner image
## Summary
`agent-ci` can fail early when `@actions/workflow-parser` does not return `template.jobs` for otherwise valid workflows (for example, workflows containing expressions the parser doesn't currently model, like `runner.temp` in env values).

The current failure mode in `handleWorkflow()` is:

- `TypeError: Cannot read properties of undefined (reading 'filter')`
- no jobs are executed

In practice this blocks local CI even when the workflow runs fine on GitHub-hosted runners.

A second practical limitation is that the runner image is hardcoded to `ghcr.io/actions/actions-runner:latest`, which forces per-repo workflow hacks when projects need local toolchain additions.

## Reproduction

1. Create a workflow with job env using `runner.temp`, e.g.:

```yaml
env:
  HOME: ${{ runner.temp }}/home
```

2. Run:

```bash
agent-ci run --workflow .github/workflows/ci.yml
```

3. Observe failure before job dispatch with `Cannot read properties of undefined (reading 'filter')`.

## Root cause

In `packages/cli/src/cli.ts`, `handleWorkflow()` assumes:

```ts
const jobs = template.jobs.filter((j) => j.type === "job")
```

If parser conversion yields an object without a `jobs` array, this throws and aborts the run.

`parseWorkflowSteps()` also assumes parsed job lookup always succeeds, which compounds parser edge-case failures.

## Proposed fix

1. **Graceful job fallback in `handleWorkflow()`**
   - If `template.jobs` is absent, read raw YAML `jobs` keys and continue.

2. **Graceful step fallback in `parseWorkflowSteps()`**
   - If parsed job lookup fails, reconstruct job/steps from raw YAML (`run`/`uses`/`with`) and continue.

3. **Configurable local runner image**
   - In `local-job.ts`, use:
     - `process.env.AGENT_CI_RUNNER_IMAGE || "ghcr.io/actions/actions-runner:latest"`
   - Allows teams to use a prebuilt local runner image without mutating workflows.

## Why this matters

- Improves compatibility with real-world workflows that GitHub accepts but parser conversion may partially miss.
- Converts hard crashes into runnable behavior with safe fallback semantics.
- Reduces per-repo local CI hacks and makes `agent-ci` easier to adopt across heterogeneous stacks.

## Notes

I have a PR ready from a fork implementing these changes with type-safe adjustments and passing package tests/build in the fork environment.

### PR DESCRIPTION
**Pull request by kaka-ruto**
## Summary
This PR improves local workflow compatibility and runner portability in 3 ways:

1. Adds a fallback in `handleWorkflow()` when parser conversion omits `template.jobs`:
   - falls back to raw YAML `jobs` keys instead of throwing on `.filter`
2. Adds a fallback in `parseWorkflowSteps()` when parsed job lookup fails:
   - reconstructs minimal job/step shape from raw YAML (`run`/`uses`/`with`)
3. Makes local runner image configurable:
   - `AGENT_CI_RUNNER_IMAGE` env var override

## Why
Some valid workflows accepted by GitHub can produce incomplete parse-converted templates in `@actions/workflow-parser` for certain expressions/context usage. The current behavior hard-fails before dispatching any job.

These fallbacks keep `agent-ci` operational with best-effort semantics and reduce repo-local hacks.

## Behavior change (validated)
- Parser omission fallback:
  - Before: workflow load could fail early with `Cannot read properties of undefined (reading 'filter')`.
  - After: when `template.jobs` is missing, `agent-ci` falls back to raw YAML jobs and continues.

- Step parsing fallback:
  - Before: parsed job lookup failures could abort step extraction.
  - After: `agent-ci` reconstructs minimal step shapes from raw YAML (`run`/`uses`/`with`) and continues.

- Runner image configurability:
  - `AGENT_CI_RUNNER_IMAGE` allows local toolchain customization without per-repo workflow mutation.

Net effect:
- Previously blocking parser edge cases now degrade gracefully and run.
- Existing standard flows continue to execute.

## Related
- Closes #96

## Validation
- `pnpm -C packages/cli test`
- `pnpm -C packages/dtu-github-actions build`
- `pnpm -C packages/cli build`

### CODE REVIEW

File: experiments/retry-proof/value.txt [hunk 1]
```diff
@@ -1 +1 @@
-pass
\ No newline at end of file
+fail
```
Comment (kaka-ruto) [inline experiments/retry-proof/value.txt]: This was necessary in order to keep the retry-proof workflow failing so we can can prove non-zero exit propagation and failed step output surfacing in summary.
